tests/test_main_views.py

```diff
@@ -15,11 +23,599 @@ def create_app(self):
         return current_app
 
     def tearDown(self):
+        models.Journal.objects.delete()
         dbsql.session.remove()
         dbsql.drop_all()
 
     def test_home_page(self):
+        """
+        Teste da página inicial, deve retorna utf-8 como conjunto de caracter e
+        o template ``collection/index.html``.
+        """
         response = self.client.get(url_for('main.index'))
         self.assertEqual(200, response.status_code)
         self.assertEqual('text/html; charset=utf-8', response.content_type)
         self.assert_template_used("collection/index.html")
+
+    def test_change_set_locale(self):
+        """
+        Teste para alterar o idioma da interface, nesse teste a URL:
+        '/set_locale/<string:lang_code>' deve criar uma variável na sessão com
+        o valor informado.
+        """
+
+        with self.create_app().test_client() as c:
+            response = c.get(url_for('main.set_locale', lang_code='es_ES'))
+            self.assertEqual(302, response.status_code)
+            self.assertEqual(flask.session['lang'], 'es_ES')
+
+    def test_change_set_locale_with_unknow_lang(self):
+        """
+        Teste para alterar o idioma da interface, nesse teste a URL:
+        '/set_locale/<string:lang_code>' deve retornar uma página com
+        ``status_code``400 e manter o idioma padrão ``pt_BR``.
+        """
+
+        with self.create_app().test_client() as c:
+            response = c.get(url_for('main.set_locale', lang_code='en_US'))
+            self.assertEqual(400, response.status_code)
+            self.assertIn('Código de idioma inválido'.decode('utf-8'),
+                          response.data.decode('utf-8'))
+
+    def test_collection_list_alpha(self):
+        """
+        Teste para avaliar o retorno da ``view function`` collection_list_alpha,
+        ao cadastrarmos 10 periódico a interface deve retornar uma listagem
+        contendo elementos esperado também deve retornar o template
+        ``collection/list_alpha.html``.
+        """
+
+        journals = utils.makeAnyJournal(items=10)
+
+        response = self.client.get(url_for('main.collection_list_alpha'))
+
+        self.assertEqual(200, response.status_code)
```

Comment:
flask testing também fornece um assert específico para status_code:
 ``self.assertStatus(response, 200)``
tal vez um Find and Replace ajude, para não ter muito trabalho.